test/module/irohad/ametsuchi/postgres_query_executor_test.cpp

```diff
@@ -1104,43 +1123,188 @@ namespace iroha {
         hash3 = txs2.at(0).hash();
       }
 
+      shared_model::crypto::Hash fake_hash{zero_string};
+      shared_model::crypto::PublicKey fake_pubkey{zero_string};
+      shared_model::crypto::Hash hash1;
+      shared_model::crypto::Hash hash2;
+      shared_model::crypto::Hash hash3;
+    };
+
+
+    template <typename QueryTxPaginationTest>
+    class GetPagedTransactionsExecutorTest
+        : public GetTransactionsExecutorTest {
+     protected:
+      using Impl = QueryTxPaginationTest;
+
       // create valid transactions and commit them
-      std::vector<shared_model::proto::Transaction> createTransactionsAndCommit(
-          size_t size,
-          const shared_model::interface::types::AccountIdType &creator) {
-        std::vector<shared_model::proto::Transaction> txs;
-        txs.reserve(size);
-        for (size_t i = 0; i < size; i++) {
-          txs.push_back(
-              TestTransactionBuilder()
-                  .creatorAccountId(account->accountId())
-                  .addAssetQuantity(asset_id, std::to_string(i + 1) + ".0")
-                  .build());
+      void createTransactionsAndCommit(size_t transactions_amount) {
+        addPerms(Impl::getUserPermissions());
+
+        auto initial_txs = Impl::makeInitialTransactions(transactions_amount);
+        auto target_txs = Impl::makeTargetTransactions(transactions_amount);
+
+        tx_hashes_.reserve(target_txs.size());
+        initial_txs.reserve(initial_txs.size() + target_txs.size());
+        for (auto &tx : target_txs) {
+          tx_hashes_.emplace_back(tx.hash());
+          initial_txs.emplace_back(std::move(tx));
         }
 
         auto block = TestBlockBuilder()
-                         .transactions(txs)
+                         .transactions(initial_txs)
                          .height(1)
                          .prevHash(fake_hash)
                          .build();
 
         apply(storage, block);
+      }
 
-        return txs;
+      auto queryPage(
+          types::TransactionsNumberType page_size,
+          const boost::optional<types::HashType> &first_hash = boost::none) {
+        auto query = Impl::makeQuery(page_size, first_hash);
+        return executeQuery(query);
       }
 
-      std::string zero_string{32, '0'};
-      shared_model::crypto::Hash fake_hash{zero_string};
-      shared_model::crypto::PublicKey fake_pubkey{zero_string};
-      const std::string asset_id = "coin#domain";
-      std::unique_ptr<shared_model::interface::Account> account2;
-      shared_model::crypto::Hash hash1;
-      shared_model::crypto::Hash hash2;
-      shared_model::crypto::Hash hash3;
+      void GeneralTransactionsPageResponseCheck(
```

Comment:
Please, add a comment with information, what exactly does this function check about page response

Reply:
Otherwise, it is very hard to understand all checks, and thus maintain them

Reply:
You are absolutely right, I wanted to but forgot.